transport/socks5/socks5.go

```diff
@@ -119,3 +127,47 @@ func appendSOCKS5Address(b []byte, address string) ([]byte, error) {
 	b = binary.BigEndian.AppendUint16(b, uint16(portNum))
 	return b, nil
 }
+
+func readAddress(reader io.Reader) (string, int, error) {
+	// Read the address type
+	// The maximum buffer size is:
+	// 1 address type + 1 address length + 256 (max domain name length)
+	var buffer [1 + 1 + 256]byte
+	addrLen := 0
+	_, err := io.ReadFull(reader, buffer[:1])
+	if err != nil {
+		return "", 0, fmt.Errorf("failed to read address type: %w", err)
+	}
+	// Read the address type
+	switch buffer[0] {
+	case addrTypeIPv4:
+		addrLen = 4
```

Comment:
Put the logic to read each type in their respective case.

Reply:
@fortuna updated readAddress accordingly.